cscs-checks/tools/profiling_and_debugging/notool.py

```diff
@@ -23,66 +21,79 @@ def __init__(self, lang):
         self.num_cpus_per_task = 4
-        self.num_iterations = 200
+        self.num_tasks_per_core = 1
+        self.use_multithreading = False
+        self.num_iterations = 100
         self.variables = {
             'OMP_NUM_THREADS': str(self.num_cpus_per_task),
             'ITERATIONS': str(self.num_iterations),
             'OMP_PROC_BIND': 'true',
             'CRAYPE_LINK_TYPE': 'dynamic'
         }
+        # OpenMP support varies between compilers:
         self.openmp_versions = ScopedDict({
             'PrgEnv-cray': {'version': 201511},
             'PrgEnv-gnu': {'version': 201511},
             'PrgEnv-intel': {'version': 201611},
             'PrgEnv-pgi': {'version': 201307},
         })
-        # a scopedict is better than this:
-        # if (self.language == 'C++' and
+        self.lang = lang
+        # The scopedict (above) is better than this:
+        # if (self.lang == 'C++' and
         #    self.current_environ.name == 'PrgEnv-pgi'):
         #    self.omp_versions['PrgEnv-pgi'] = '200805'
-
+        self.maintainers = ['JG', 'MK']
+        self.tags = {'production'}
+        if self.current_system.name in {'dom', 'daint'}:
+            # get general info about the environment:
+            self.post_run = ['module list -t']
         self.perf_patterns = {
             'elapsed_time': sn.extractsingle(r'Elapsed Time\s*:\s+(\S+)',
                                              self.stdout, 1, float)
         }
-        self.reference_prgenv = {
-            'PrgEnv-gnu': (0.90, -0.6, None),
-            'PrgEnv-cray': (0.90, -0.6, None),
-            'PrgEnv-intel': (0.90, -0.6, None),
-            'PrgEnv-pgi': (18.0, -0.6, None),
-        }
         self.reference = {
             '*': {
-                'elapsed_time': (0, None, None)
+                'elapsed_time': (0, None, None, 'seconds')
             }
         }
-        self.post_run = ['module list -t']
-
-        self.maintainers = ['JG', 'MK']
-        self.tags = {'production'}
+        if lang == 'C++':
+            self.reference_prgenv = {
+                'PrgEnv-cray': (0.38, -0.6, None, 'seconds'),
+                'PrgEnv-gnu': (0.38, -0.6, None, 'seconds'),
+                'PrgEnv-intel': (0.38, -0.6, None, 'seconds'),
+                'PrgEnv-pgi': (18.0, -0.6, None, 'seconds'),
+            }
+        elif lang == 'F90':
+            self.reference_prgenv = {
+                'PrgEnv-cray': (0.17, -0.6, None, 'seconds'),
+                'PrgEnv-gnu': (0.17, -0.6, None, 'seconds'),
+                'PrgEnv-intel': (0.17, -0.6, None, 'seconds'),
+                'PrgEnv-pgi': (18.0, -0.6, None, 'seconds'),
+            }
 
     def setup(self, partition, environ, **job_opts):
         super().setup(partition, environ, **job_opts)
-        environ_name = self.current_environ.name
-        prgenv_flags = self.prgenv_flags[environ_name]
+        envname = self.current_environ.name
+        prgenv_flags = self.prgenv_flags[envname]
```

Comment:
This is not generic. What if sb tries it on a system with a different prgenv? This will make the test crash. You should write it as follows:

```python
prgenv_flags = self.prgenv_flags.get(envname, ['put', 'some', 'default', 'flags', 'if', 'any'])
```